apple/swift-docc
# encode overload declaration references into render references

Bug/issue #, if applicable: rdar://117805293

## Summary

The overload declaration collection done in #740 failed to add the overload render references to the render node, meaning that Swift-DocC-Render can't navigate to the pages for other overloads if they're not already referenced in the prose or automatic curation. This PR ensures that these references are added to the render node.

## Dependencies

None

## Testing

Steps:
1. With a recent `main` branch Swift-DocC-Render built in `$DOCC_HTML_DIR`, run `swift run docc preview --enable-experimental-overloaded-symbol-presentation Tests/SwiftDocCTests/Test\ Bundles/OverloadedSymbols.docc`
2. Navigate to `OverloadedEnum/firstTestMemberName(_:)` (any overload of that method)
3. Expand the "Show all declarations" listing, and click a different overload's declaration.
4. Ensure that the other overload's page is loaded, both by the URL changing and by the sidebar navigator changing to highlight the selected overload.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ n/a ] Updated documentation if necessary
